lib/services/tokens/token.dao.js

```diff
@@ -72,27 +69,31 @@ dao.find = function (tokenObj, options) {
         return true;
       });
 
-      tokenPromises = activeTokenIds.map((id) => {
-        return db.hgetallAsync(tokenNamespace.concat(':', id))
-          .then((token) => {
-            let isEqual;
-
-            if (!token) {
-              return Promise.reject(new Error());
-            }
+      const getTokenPromise = new Promise((resolve, reject) => {
+        activeTokenIds.reverse();
 
-            isEqual = Object.keys(tokenObj).every((key) => tokenObj[key] === token[key]);
-            return isEqual ? token : Promise.reject(new Error());
-          });
+        const next = () => {
+          if (!activeTokenIds.length) {
+            reject(new Error());
+            return;
+          }
+          db.hgetallAsync(tokenNamespace.concat(':', activeTokenIds.pop()))
+            .then((token) => {
+              if (!token || !Object.keys(tokenObj).every((key) => tokenObj[key] === token[key])) {
+                throw new Error();
```

Comment:
I'd put an error message here.

Reply:
new Error() was here before. Take a look what happened with that error next (`catch(() => null)`). In my opinion whole file needs to be rewritten.